libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1064,19 +1073,24 @@ string YulUtilFunctions::clearStorageRangeFunction(Type const& _type)
 {
 	string functionName = "clear_storage_range_" + _type.identifier();
 
-	solAssert(_type.storageBytes() >= 32, "Expected smaller value for storage bytes");
-
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
-			function <functionName>(start, end) {
+			function <functionName>(start, offset, end) {
```

Comment:
Actually I'm wondering if it might be better to use a second function just to clear the single part of the first slot and then call the already existing function

Reply:
Or at least somehow make sure that for types for which offset is always zero, this code is not even generated.

Reply:
Created another function that clears storage slot partially, and called from `<?packed>` part.